From e1a03ead7a541a2118816b1632e957cc2bb44c8f Mon Sep 17 00:00:00 2001
From: Benjamin Otte <otte@redhat.com>
Date: Mon, 29 Aug 2016 16:00:17 +0200
Subject: Use NULL for generic marshallers in g_signal_new()

glib will use the correct marshaller automatically. And as a side
effect, we also get all glib optimizations, like a va marshaller.
---
 gtk/deprecated/gtkactiongroup.c |  4 ++--
 gtk/deprecated/gtkcolorsel.c    |  2 +-
 gtk/deprecated/gtkhandlebox.c   |  4 ++--
 gtk/deprecated/gtkhsv.c         |  4 ++--
 gtk/deprecated/gtkstyle.c       |  4 ++--
 gtk/deprecated/gtkuimanager.c   |  4 ++--
 gtk/gtkadjustment.c             |  4 ++--
 gtk/gtkappchooserbutton.c       |  2 +-
 gtk/gtkappchooserwidget.c       |  4 ++--
 gtk/gtkbutton.c                 | 12 +++++-----
 gtk/gtkcalendar.c               | 14 ++++++------
 gtk/gtkcelleditable.c           |  4 ++--
 gtk/gtkcellrenderer.c           |  2 +-
 gtk/gtkcellrenderertoggle.c     |  2 +-
 gtk/gtkcheckmenuitem.c          |  2 +-
 gtk/gtkclipboard-quartz.c       |  2 +-
 gtk/gtkclipboard.c              |  2 +-
 gtk/gtkcolorbutton.c            |  2 +-
 gtk/gtkcontainer.c              |  8 +++----
 gtk/gtkcssnode.c                |  5 +----
 gtk/gtkdialog.c                 |  4 ++--
 gtk/gtkeditable.c               |  2 +-
 gtk/gtkentry.c                  | 18 +++++++--------
 gtk/gtkentrycompletion.c        |  2 +-
 gtk/gtkexpander.c               |  2 +-
 gtk/gtkfilechooserbutton.c      |  2 +-
 gtk/gtkfilechooserembed.c       |  4 ++--
 gtk/gtkfilechooserentry.c       |  2 +-
 gtk/gtkfilechooserwidget.c      | 24 ++++++++++----------
 gtk/gtkfilesystemmodel.c        |  2 +-
 gtk/gtkimcontext.c              |  8 +++----
 gtk/gtklabel.c                  |  6 ++---
 gtk/gtklevelbar.c               |  2 +-
 gtk/gtklistbox.c                |  2 +-
 gtk/gtkmarshalers.list          | 10 ---------
 gtk/gtkmenu.c                   |  2 +-
 gtk/gtkmenuitem.c               | 12 +++++-----
 gtk/gtkmenushell.c              | 12 +++++-----
 gtk/gtknotebook.c               |  2 +-
 gtk/gtkplacessidebar.c          | 12 +++++-----
 gtk/gtkplug.c                   |  2 +-
 gtk/gtkradiobutton.c            |  2 +-
 gtk/gtkradiomenuitem.c          |  2 +-
 gtk/gtkrange.c                  |  6 ++---
 gtk/gtkscalebutton.c            |  2 +-
 gtk/gtkscrolledwindow.c         |  2 +-
 gtk/gtksearchentry.c            |  8 +++----
 gtk/gtksocket.c                 |  2 +-
 gtk/gtkspinbutton.c             |  6 ++---
 gtk/gtkswitch.c                 |  2 +-
 gtk/gtktextbuffer.c             | 12 +++++-----
 gtk/gtktextlayout.c             |  2 +-
 gtk/gtktexttagtable.c           |  4 ++--
 gtk/gtktextview.c               | 22 +++++++++---------
 gtk/gtktogglebutton.c           |  2 +-
 gtk/gtktoolitem.c               |  2 +-
 gtk/gtktrashmonitor.c           |  2 +-
 gtk/gtktreemenu.c               |  2 +-
 gtk/gtktreemodel.c              |  2 +-
 gtk/gtktreeselection.c          |  2 +-
 gtk/gtktreesortable.c           |  2 +-
 gtk/gtktreeview.c               |  4 ++--
 gtk/gtktreeviewcolumn.c         |  2 +-
 gtk/gtkwidget.c                 | 50 ++++++++++++++++++++---------------------
 gtk/gtkwindow.c                 |  8 +++----
 gtk/inspector/strv-editor.c     |  2 +-
 66 files changed, 178 insertions(+), 193 deletions(-)

diff --git a/gtk/deprecated/gtkactiongroup.c b/gtk/deprecated/gtkactiongroup.c
index 32b913c..45eeb1c 100644
--- a/gtk/deprecated/gtkactiongroup.c
+++ b/gtk/deprecated/gtkactiongroup.c
@@ -318,7 +318,7 @@ gtk_action_group_class_init (GtkActionGroupClass *klass)
     g_signal_new (I_("pre-activate"),
 		  G_OBJECT_CLASS_TYPE (klass),
 		  0, 0, NULL, NULL,
-		  _gtk_marshal_VOID__OBJECT,
+		  NULL,
 		  G_TYPE_NONE, 1, 
 		  GTK_TYPE_ACTION);
 
@@ -341,7 +341,7 @@ gtk_action_group_class_init (GtkActionGroupClass *klass)
     g_signal_new (I_("post-activate"),
 		  G_OBJECT_CLASS_TYPE (klass),
 		  0, 0, NULL, NULL,
-		  _gtk_marshal_VOID__OBJECT,
+		  NULL,
 		  G_TYPE_NONE, 1, 
 		  GTK_TYPE_ACTION);
 }
diff --git a/gtk/deprecated/gtkcolorsel.c b/gtk/deprecated/gtkcolorsel.c
index 27be9c1..8152983 100644
--- a/gtk/deprecated/gtkcolorsel.c
+++ b/gtk/deprecated/gtkcolorsel.c
@@ -384,7 +384,7 @@ gtk_color_selection_class_init (GtkColorSelectionClass *klass)
                   G_SIGNAL_RUN_FIRST,
                   G_STRUCT_OFFSET (GtkColorSelectionClass, color_changed),
                   NULL, NULL,
-                  _gtk_marshal_VOID__VOID,
+                  NULL,
                   G_TYPE_NONE, 0);
 }
 
diff --git a/gtk/deprecated/gtkhandlebox.c b/gtk/deprecated/gtkhandlebox.c
index f254138..cb1d12c 100644
--- a/gtk/deprecated/gtkhandlebox.c
+++ b/gtk/deprecated/gtkhandlebox.c
@@ -296,7 +296,7 @@ gtk_handle_box_class_init (GtkHandleBoxClass *class)
 		  G_SIGNAL_RUN_FIRST,
 		  G_STRUCT_OFFSET (GtkHandleBoxClass, child_attached),
 		  NULL, NULL,
-		  _gtk_marshal_VOID__OBJECT,
+		  NULL,
 		  G_TYPE_NONE, 1,
 		  GTK_TYPE_WIDGET);
 
@@ -318,7 +318,7 @@ gtk_handle_box_class_init (GtkHandleBoxClass *class)
 		  G_SIGNAL_RUN_FIRST,
 		  G_STRUCT_OFFSET (GtkHandleBoxClass, child_detached),
 		  NULL, NULL,
-		  _gtk_marshal_VOID__OBJECT,
+		  NULL,
 		  G_TYPE_NONE, 1,
 		  GTK_TYPE_WIDGET);
 }
diff --git a/gtk/deprecated/gtkhsv.c b/gtk/deprecated/gtkhsv.c
index 2da78cc..6e30d5e 100644
--- a/gtk/deprecated/gtkhsv.c
+++ b/gtk/deprecated/gtkhsv.c
@@ -170,7 +170,7 @@ gtk_hsv_class_init (GtkHSVClass *class)
                   G_SIGNAL_RUN_FIRST,
                   G_STRUCT_OFFSET (GtkHSVClass, changed),
                   NULL, NULL,
-                  _gtk_marshal_VOID__VOID,
+                  NULL,
                   G_TYPE_NONE, 0);
 
   hsv_signals[MOVE] =
@@ -179,7 +179,7 @@ gtk_hsv_class_init (GtkHSVClass *class)
                   G_SIGNAL_RUN_LAST | G_SIGNAL_ACTION,
                   G_STRUCT_OFFSET (GtkHSVClass, move),
                   NULL, NULL,
-                  _gtk_marshal_VOID__ENUM,
+                  NULL,
                   G_TYPE_NONE, 1,
                   GTK_TYPE_DIRECTION_TYPE);
 
diff --git a/gtk/deprecated/gtkstyle.c b/gtk/deprecated/gtkstyle.c
index 403c460..291e5c2 100644
--- a/gtk/deprecated/gtkstyle.c
+++ b/gtk/deprecated/gtkstyle.c
@@ -489,7 +489,7 @@ gtk_style_class_init (GtkStyleClass *kla
 				 G_SIGNAL_RUN_FIRST,
 				 G_STRUCT_OFFSET (GtkStyleClass, realize),
 				 NULL, NULL,
-				 _gtk_marshal_VOID__VOID,
+				 NULL,
 				 G_TYPE_NONE, 0);
   /**
    * GtkStyle::unrealize:
@@ -507,7 +507,7 @@ gtk_style_class_init (GtkStyleClass *kla
 				   G_SIGNAL_RUN_FIRST,
 				   G_STRUCT_OFFSET (GtkStyleClass, unrealize),
 				   NULL, NULL,
-				   _gtk_marshal_VOID__VOID,
+				   NULL,
 				   G_TYPE_NONE, 0);
 
   quark_default_style = g_quark_from_static_string ("gtk-legacy-default-style");
diff --git a/gtk/deprecated/gtkuimanager.c b/gtk/deprecated/gtkuimanager.c
index 9eea97b..a0b4129 100644
--- a/gtk/deprecated/gtkuimanager.c
+++ b/gtk/deprecated/gtkuimanager.c
@@ -584,7 +584,7 @@ gtk_ui_manager_class_init (GtkUIManagerClass *klass)
 		  G_SIGNAL_RUN_FIRST | G_SIGNAL_NO_RECURSE,
 		  G_STRUCT_OFFSET (GtkUIManagerClass, pre_activate),
 		  NULL, NULL,
-		  _gtk_marshal_VOID__OBJECT,
+		  NULL,
 		  G_TYPE_NONE, 1,
 		  GTK_TYPE_ACTION);
 
@@ -609,7 +609,7 @@ gtk_ui_manager_class_init (GtkUIManagerClass *klass)
 		  G_SIGNAL_RUN_FIRST | G_SIGNAL_NO_RECURSE,
 		  G_STRUCT_OFFSET (GtkUIManagerClass, post_activate),
 		  NULL, NULL,
-		  _gtk_marshal_VOID__OBJECT,
+		  NULL,
 		  G_TYPE_NONE, 1,
 		  GTK_TYPE_ACTION);
 
diff --git a/gtk/gtkadjustment.c b/gtk/gtkadjustment.c
index 4c0f6af..423f88f 100644
--- a/gtk/gtkadjustment.c
+++ b/gtk/gtkadjustment.c
@@ -241,7 +241,7 @@ gtk_adjustment_class_init (GtkAdjustmentClass *class)
 		  G_SIGNAL_RUN_FIRST | G_SIGNAL_NO_RECURSE,
 		  G_STRUCT_OFFSET (GtkAdjustmentClass, changed),
 		  NULL, NULL,
-		  _gtk_marshal_VOID__VOID,
+		  NULL,
 		  G_TYPE_NONE, 0);
 
   /**
@@ -256,7 +256,7 @@ gtk_adjustment_class_init (GtkAdjustmentClass *class)
 		  G_SIGNAL_RUN_FIRST | G_SIGNAL_NO_RECURSE,
 		  G_STRUCT_OFFSET (GtkAdjustmentClass, value_changed),
 		  NULL, NULL,
-		  _gtk_marshal_VOID__VOID,
+		  NULL,
 		  G_TYPE_NONE, 0);
 }
 
diff --git a/gtk/gtkappchooserbutton.c b/gtk/gtkappchooserbutton.c
index 7b6dcff..c0af584 100644
--- a/gtk/gtkappchooserbutton.c
+++ b/gtk/gtkappchooserbutton.c
@@ -666,7 +666,7 @@ gtk_app_chooser_button_class_init (GtkAppChooserButtonClass *klass)
                   G_SIGNAL_RUN_FIRST | G_SIGNAL_DETAILED,
                   G_STRUCT_OFFSET (GtkAppChooserButtonClass, custom_item_activated),
                   NULL, NULL,
-                  _gtk_marshal_VOID__STRING,
+                  NULL,
                   G_TYPE_NONE,
                   1, G_TYPE_STRING);
 }
diff --git a/gtk/gtkappchooserwidget.c b/gtk/gtkappchooserwidget.c
index fd9e6b9..f45348e 100644
--- a/gtk/gtkappchooserwidget.c
+++ b/gtk/gtkappchooserwidget.c
@@ -1050,7 +1050,7 @@ gtk_app_chooser_widget_class_init (GtkAppChooserWidgetClass *klass)
                   G_SIGNAL_RUN_FIRST,
                   G_STRUCT_OFFSET (GtkAppChooserWidgetClass, application_selected),
                   NULL, NULL,
-                  _gtk_marshal_VOID__OBJECT,
+                  NULL,
                   G_TYPE_NONE,
                   1, G_TYPE_APP_INFO);
 
@@ -1071,7 +1071,7 @@ gtk_app_chooser_widget_class_init (GtkAppChooserWidgetClass *klass)
                   G_SIGNAL_RUN_FIRST,
                   G_STRUCT_OFFSET (GtkAppChooserWidgetClass, application_activated),
                   NULL, NULL,
-                  _gtk_marshal_VOID__OBJECT,
+                  NULL,
                   G_TYPE_NONE,
                   1, G_TYPE_APP_INFO);
 
diff --git a/gtk/gtkbutton.c b/gtk/gtkbutton.c
index 8c570a9..9b645c3 100644
--- a/gtk/gtkbutton.c
+++ b/gtk/gtkbutton.c
@@ -419,7 +419,7 @@ gtk_button_class_init (GtkButtonClass *klass)
 		  G_SIGNAL_RUN_FIRST,
 		  G_STRUCT_OFFSET (GtkButtonClass, pressed),
 		  NULL, NULL,
-		  _gtk_marshal_VOID__VOID,
+		  NULL,
 		  G_TYPE_NONE, 0);
 
   /**
@@ -436,7 +436,7 @@ gtk_button_class_init (GtkButtonClass *klass)
 		  G_SIGNAL_RUN_FIRST,
 		  G_STRUCT_OFFSET (GtkButtonClass, released),
 		  NULL, NULL,
-		  _gtk_marshal_VOID__VOID,
+		  NULL,
 		  G_TYPE_NONE, 0);
 
   /**
@@ -451,7 +451,7 @@ gtk_button_class_init (GtkButtonClass *klass)
 		  G_SIGNAL_RUN_FIRST | G_SIGNAL_ACTION,
 		  G_STRUCT_OFFSET (GtkButtonClass, clicked),
 		  NULL, NULL,
-		  _gtk_marshal_VOID__VOID,
+		  NULL,
 		  G_TYPE_NONE, 0);
 
   /**
@@ -468,7 +468,7 @@ gtk_button_class_init (GtkButtonClass *klass)
 		  G_SIGNAL_RUN_FIRST,
 		  G_STRUCT_OFFSET (GtkButtonClass, enter),
 		  NULL, NULL,
-		  _gtk_marshal_VOID__VOID,
+		  NULL,
 		  G_TYPE_NONE, 0);
 
   /**
@@ -485,7 +485,7 @@ gtk_button_class_init (GtkButtonClass *klass)
 		  G_SIGNAL_RUN_FIRST,
 		  G_STRUCT_OFFSET (GtkButtonClass, leave),
 		  NULL, NULL,
-		  _gtk_marshal_VOID__VOID,
+		  NULL,
 		  G_TYPE_NONE, 0);
 
   /**
@@ -503,7 +503,7 @@ gtk_button_class_init (GtkButtonClass *klass)
 		  G_SIGNAL_RUN_FIRST | G_SIGNAL_ACTION,
 		  G_STRUCT_OFFSET (GtkButtonClass, activate),
 		  NULL, NULL,
-		  _gtk_marshal_VOID__VOID,
+		  NULL,
 		  G_TYPE_NONE, 0);
   widget_class->activate_signal = button_signals[ACTIVATE];
 
diff --git a/gtk/gtkcalendar.c b/gtk/gtkcalendar.c
index da319b1..f274c63 100644
--- a/gtk/gtkcalendar.c
+++ b/gtk/gtkcalendar.c
@@ -587,7 +587,7 @@ gtk_calendar_class_init (GtkCalendarClass *class)
                   G_SIGNAL_RUN_FIRST,
                   G_STRUCT_OFFSET (GtkCalendarClass, month_changed),
                   NULL, NULL,
-                  _gtk_marshal_VOID__VOID,
+                  NULL,
                   G_TYPE_NONE, 0);
 
   /**
@@ -602,7 +602,7 @@ gtk_calendar_class_init (GtkCalendarClass *class)
                   G_SIGNAL_RUN_FIRST,
                   G_STRUCT_OFFSET (GtkCalendarClass, day_selected),
                   NULL, NULL,
-                  _gtk_marshal_VOID__VOID,
+                  NULL,
                   G_TYPE_NONE, 0);
 
   /**
@@ -617,7 +617,7 @@ gtk_calendar_class_init (GtkCalendarClass *class)
                   G_SIGNAL_RUN_FIRST,
                   G_STRUCT_OFFSET (GtkCalendarClass, day_selected_double_click),
                   NULL, NULL,
-                  _gtk_marshal_VOID__VOID,
+                  NULL,
                   G_TYPE_NONE, 0);
 
   /**
@@ -632,7 +632,7 @@ gtk_calendar_class_init (GtkCalendarClass *class)
                   G_SIGNAL_RUN_FIRST,
                   G_STRUCT_OFFSET (GtkCalendarClass, prev_month),
                   NULL, NULL,
-                  _gtk_marshal_VOID__VOID,
+                  NULL,
                   G_TYPE_NONE, 0);
 
   /**
@@ -647,7 +647,7 @@ gtk_calendar_class_init (GtkCalendarClass *class)
                   G_SIGNAL_RUN_FIRST,
                   G_STRUCT_OFFSET (GtkCalendarClass, next_month),
                   NULL, NULL,
-                  _gtk_marshal_VOID__VOID,
+                  NULL,
                   G_TYPE_NONE, 0);
 
   /**
@@ -662,7 +662,7 @@ gtk_calendar_class_init (GtkCalendarClass *class)
                   G_SIGNAL_RUN_FIRST,
                   G_STRUCT_OFFSET (GtkCalendarClass, prev_year),
                   NULL, NULL,
-                  _gtk_marshal_VOID__VOID,
+                  NULL,
                   G_TYPE_NONE, 0);
 
   /**
@@ -677,7 +677,7 @@ gtk_calendar_class_init (GtkCalendarClass *class)
                   G_SIGNAL_RUN_FIRST,
                   G_STRUCT_OFFSET (GtkCalendarClass, next_year),
                   NULL, NULL,
-                  _gtk_marshal_VOID__VOID,
+                  NULL,
                   G_TYPE_NONE, 0);
 
   gtk_widget_class_set_accessible_role (widget_class, ATK_ROLE_CALENDAR);
diff --git a/gtk/gtkcelleditable.c b/gtk/gtkcelleditable.c
index 1c9e2eb..1b02665 100644
--- a/gtk/gtkcelleditable.c
+++ b/gtk/gtkcelleditable.c
@@ -72,7 +72,7 @@ gtk_cell_editable_default_init (GtkCellEditableInterface *iface)
                 G_SIGNAL_RUN_LAST,
                 G_STRUCT_OFFSET (GtkCellEditableIface, editing_done),
                 NULL, NULL,
-                _gtk_marshal_VOID__VOID,
+                NULL,
                 G_TYPE_NONE, 0);
 
   /**
@@ -96,7 +96,7 @@ gtk_cell_editable_default_init (GtkCellEditableInterface *iface)
                 G_SIGNAL_RUN_LAST,
                 G_STRUCT_OFFSET (GtkCellEditableIface, remove_widget),
                 NULL, NULL,
-                _gtk_marshal_VOID__VOID,
+                NULL,
                 G_TYPE_NONE, 0);
 }
 
diff --git a/gtk/gtkcellrenderer.c b/gtk/gtkcellrenderer.c
index ce243b0..58969d4 100644
--- a/gtk/gtkcellrenderer.c
+++ b/gtk/gtkcellrenderer.c
@@ -230,7 +230,7 @@ gtk_cell_renderer_class_init (GtkCellRendererClass *class)
 		  G_SIGNAL_RUN_FIRST,
 		  G_STRUCT_OFFSET (GtkCellRendererClass, editing_canceled),
 		  NULL, NULL,
-		  _gtk_marshal_VOID__VOID,
+		  NULL,
 		  G_TYPE_NONE, 0);
 
   /**
diff --git a/gtk/gtkcellrenderertoggle.c b/gtk/gtkcellrenderertoggle.c
index fcf82ab..ea5aebf 100644
--- a/gtk/gtkcellrenderertoggle.c
+++ b/gtk/gtkcellrenderertoggle.c
@@ -193,7 +193,7 @@ gtk_cell_renderer_toggle_class_init (GtkCellRendererToggleClass *class)
 		  G_SIGNAL_RUN_LAST,
 		  G_STRUCT_OFFSET (GtkCellRendererToggleClass, toggled),
 		  NULL, NULL,
-		  _gtk_marshal_VOID__STRING,
+		  NULL,
 		  G_TYPE_NONE, 1,
 		  G_TYPE_STRING);
 
diff --git a/gtk/gtkcheckmenuitem.c b/gtk/gtkcheckmenuitem.c
index 4a0f8ff..b3b32d2 100644
--- a/gtk/gtkcheckmenuitem.c
+++ b/gtk/gtkcheckmenuitem.c
@@ -268,7 +268,7 @@ gtk_check_menu_item_class_init (GtkCheckMenuItemClass *klass)
                   G_SIGNAL_RUN_FIRST,
                   G_STRUCT_OFFSET (GtkCheckMenuItemClass, toggled),
                   NULL, NULL,
-                  _gtk_marshal_VOID__VOID,
+                  NULL,
                   G_TYPE_NONE, 0);
 
   gtk_widget_class_set_accessible_type (widget_class, GTK_TYPE_CHECK_MENU_ITEM_ACCESSIBLE);
diff --git a/gtk/gtkclipboard-quartz.c b/gtk/gtkclipboard-quartz.c
index 8f98cc7..a51c46d 100644
--- a/gtk/gtkclipboard-quartz.c
+++ b/gtk/gtkclipboard-quartz.c
@@ -209,7 +209,7 @@ gtk_clipboard_class_init (GtkClipboardClass *class)
 		  G_SIGNAL_RUN_FIRST,
 		  G_STRUCT_OFFSET (GtkClipboardClass, owner_change),
 		  NULL, NULL,
-		  _gtk_marshal_VOID__BOXED,
+		  NULL,
 		  G_TYPE_NONE, 1,
 		  GDK_TYPE_EVENT | G_SIGNAL_TYPE_STATIC_SCOPE);
 }
diff --git a/gtk/gtkclipboard.c b/gtk/gtkclipboard.c
index ebd9f4e..06a8a39 100644
--- a/gtk/gtkclipboard.c
+++ b/gtk/gtkclipboard.c
@@ -245,7 +245,7 @@ gtk_clipboard_class_init (GtkClipboardClass *class)
 		  G_SIGNAL_RUN_FIRST,
 		  G_STRUCT_OFFSET (GtkClipboardClass, owner_change),
 		  NULL, NULL,
-		  _gtk_marshal_VOID__BOXED,
+		  NULL,
 		  G_TYPE_NONE, 1,
 		  GDK_TYPE_EVENT | G_SIGNAL_TYPE_STATIC_SCOPE);
 }
diff --git a/gtk/gtkcolorbutton.c b/gtk/gtkcolorbutton.c
index 060f8a1..b77ea28 100644
--- a/gtk/gtkcolorbutton.c
+++ b/gtk/gtkcolorbutton.c
@@ -254,7 +254,7 @@ G_GNUC_END_IGNORE_DEPRECATIONS
                                                   G_SIGNAL_RUN_FIRST,
                                                   G_STRUCT_OFFSET (GtkColorButtonClass, color_set),
                                                   NULL, NULL,
-                                                  _gtk_marshal_VOID__VOID,
+                                                  NULL,
                                                   G_TYPE_NONE, 0);
 
   /**
diff --git a/gtk/gtkcontainer.c b/gtk/gtkcontainer.c
index 44c6573..0fbfe9a 100644
--- a/gtk/gtkcontainer.c
+++ b/gtk/gtkcontainer.c
@@ -557,7 +557,7 @@ gtk_container_class_init (GtkContainerClass *class)
                   G_SIGNAL_RUN_FIRST,
                   G_STRUCT_OFFSET (GtkContainerClass, add),
                   NULL, NULL,
-                  _gtk_marshal_VOID__OBJECT,
+                  NULL,
                   G_TYPE_NONE, 1,
                   GTK_TYPE_WIDGET);
   container_signals[REMOVE] =
@@ -566,7 +566,7 @@ gtk_container_class_init (GtkContainerClass *class)
                   G_SIGNAL_RUN_FIRST,
                   G_STRUCT_OFFSET (GtkContainerClass, remove),
                   NULL, NULL,
-                  _gtk_marshal_VOID__OBJECT,
+                  NULL,
                   G_TYPE_NONE, 1,
                   GTK_TYPE_WIDGET);
   container_signals[CHECK_RESIZE] =
@@ -575,7 +575,7 @@ gtk_container_class_init (GtkContainerClass *class)
                   G_SIGNAL_RUN_LAST,
                   G_STRUCT_OFFSET (GtkContainerClass, check_resize),
                   NULL, NULL,
-                  _gtk_marshal_VOID__VOID,
+                  NULL,
                   G_TYPE_NONE, 0);
   container_signals[SET_FOCUS_CHILD] =
     g_signal_new (I_("set-focus-child"),
@@ -583,7 +583,7 @@ gtk_container_class_init (GtkContainerClass *class)
                   G_SIGNAL_RUN_FIRST,
                   G_STRUCT_OFFSET (GtkContainerClass, set_focus_child),
                   NULL, NULL,
-                  _gtk_marshal_VOID__OBJECT,
+                  NULL,
                   G_TYPE_NONE, 1,
                   GTK_TYPE_WIDGET);
 
diff --git a/gtk/gtkcssnode.c b/gtk/gtkcssnode.c
index 917cd56..5be2806 100644
--- a/gtk/gtkcssnode.c
+++ b/gtk/gtkcssnode.c
@@ -616,12 +616,9 @@ gtk_css_node_class_init (GtkCssNodeClass *klass)
 		  G_SIGNAL_RUN_LAST,
                   G_STRUCT_OFFSET (GtkCssNodeClass, style_changed),
 		  NULL, NULL,
-		  _gtk_marshal_VOID__POINTER,
+		  NULL,
 		  G_TYPE_NONE, 1,
 		  G_TYPE_POINTER);
-  g_signal_set_va_marshaller (cssnode_signals[STYLE_CHANGED],
-                              G_TYPE_FROM_CLASS (klass),
-                              _gtk_marshal_VOID__POINTERv);
 
   cssnode_properties[PROP_CLASSES] =
     g_param_spec_boxed ("classes", P_("Style Classes"), P_("List of classes"),
diff --git a/gtk/gtkdialog.c b/gtk/gtkdialog.c
index 45dc5be..85e6f75 100644
--- a/gtk/gtkdialog.c
+++ b/gtk/gtkdialog.c
@@ -552,7 +552,7 @@ gtk_dialog_class_init (GtkDialogClass *class)
 		  G_SIGNAL_RUN_LAST,
 		  G_STRUCT_OFFSET (GtkDialogClass, response),
 		  NULL, NULL,
-		  _gtk_marshal_VOID__INT,
+		  NULL,
 		  G_TYPE_NONE, 1,
 		  G_TYPE_INT);
 
@@ -572,7 +572,7 @@ gtk_dialog_class_init (GtkDialogClass *class)
 		  G_SIGNAL_RUN_LAST | G_SIGNAL_ACTION,
 		  G_STRUCT_OFFSET (GtkDialogClass, close),
 		  NULL, NULL,
-		  _gtk_marshal_VOID__VOID,
+		  NULL,
 		  G_TYPE_NONE, 0);
 
   /**
diff --git a/gtk/gtkeditable.c b/gtk/gtkeditable.c
index c0e17da..50d0df4 100644
--- a/gtk/gtkeditable.c
+++ b/gtk/gtkeditable.c
@@ -176,7 +176,7 @@ gtk_editable_base_init (gpointer g_class)
 		    G_SIGNAL_RUN_LAST,
 		    G_STRUCT_OFFSET (GtkEditableInterface, changed),
 		    NULL, NULL,
-		    _gtk_marshal_VOID__VOID,
+		    NULL,
 		    G_TYPE_NONE, 0);
 
       initialized = TRUE;
diff --git a/gtk/gtkentry.c b/gtk/gtkentry.c
index 530b551..35f9e69 100644
--- a/gtk/gtkentry.c
+++ b/gtk/gtkentry.c
@@ -1594,7 +1594,7 @@ gtk_entry_class_init (GtkEntryClass *class)
 		  G_SIGNAL_RUN_LAST,
 		  G_STRUCT_OFFSET (GtkEntryClass, populate_popup),
 		  NULL, NULL,
-		  _gtk_marshal_VOID__OBJECT,
+		  NULL,
 		  G_TYPE_NONE, 1,
 		  GTK_TYPE_WIDGET);
   
@@ -1620,7 +1620,7 @@ gtk_entry_class_init (GtkEntryClass *class)
 		  G_SIGNAL_RUN_LAST | G_SIGNAL_ACTION,
 		  G_STRUCT_OFFSET (GtkEntryClass, activate),
 		  NULL, NULL,
-		  _gtk_marshal_VOID__VOID,
+		  NULL,
 		  G_TYPE_NONE, 0);
   widget_class->activate_signal = signals[ACTIVATE];
 
@@ -1679,7 +1679,7 @@ gtk_entry_class_init (GtkEntryClass *class)
 		  G_SIGNAL_RUN_LAST | G_SIGNAL_ACTION,
 		  G_STRUCT_OFFSET (GtkEntryClass, insert_at_cursor),
 		  NULL, NULL,
-		  _gtk_marshal_VOID__STRING,
+		  NULL,
 		  G_TYPE_NONE, 1,
 		  G_TYPE_STRING);
 
@@ -1729,7 +1729,7 @@ gtk_entry_class_init (GtkEntryClass *class)
 		  G_SIGNAL_RUN_LAST | G_SIGNAL_ACTION,
 		  G_STRUCT_OFFSET (GtkEntryClass, backspace),
 		  NULL, NULL,
-		  _gtk_marshal_VOID__VOID,
+		  NULL,
 		  G_TYPE_NONE, 0);
 
   /**
@@ -1749,7 +1749,7 @@ gtk_entry_class_init (GtkEntryClass *class)
 		  G_SIGNAL_RUN_LAST | G_SIGNAL_ACTION,
 		  G_STRUCT_OFFSET (GtkEntryClass, cut_clipboard),
 		  NULL, NULL,
-		  _gtk_marshal_VOID__VOID,
+		  NULL,
 		  G_TYPE_NONE, 0);
 
   /**
@@ -1769,7 +1769,7 @@ gtk_entry_class_init (GtkEntryClass *class)
 		  G_SIGNAL_RUN_LAST | G_SIGNAL_ACTION,
 		  G_STRUCT_OFFSET (GtkEntryClass, copy_clipboard),
 		  NULL, NULL,
-		  _gtk_marshal_VOID__VOID,
+		  NULL,
 		  G_TYPE_NONE, 0);
 
   /**
@@ -1790,7 +1790,7 @@ gtk_entry_class_init (GtkEntryClass *class)
 		  G_SIGNAL_RUN_LAST | G_SIGNAL_ACTION,
 		  G_STRUCT_OFFSET (GtkEntryClass, paste_clipboard),
 		  NULL, NULL,
-		  _gtk_marshal_VOID__VOID,
+		  NULL,
 		  G_TYPE_NONE, 0);
 
   /**
@@ -1809,7 +1809,7 @@ gtk_entry_class_init (GtkEntryClass *class)
 		  G_SIGNAL_RUN_LAST | G_SIGNAL_ACTION,
 		  G_STRUCT_OFFSET (GtkEntryClass, toggle_overwrite),
 		  NULL, NULL,
-		  _gtk_marshal_VOID__VOID,
+		  NULL,
 		  G_TYPE_NONE, 0);
 
   /**
@@ -1873,7 +1873,7 @@ gtk_entry_class_init (GtkEntryClass *class)
                                 G_SIGNAL_RUN_LAST | G_SIGNAL_ACTION,
                                 NULL,
                                 NULL, NULL,
-                                _gtk_marshal_VOID__STRING,
+                                NULL,
                                 G_TYPE_NONE, 1,
                                 G_TYPE_STRING);
 
diff --git a/gtk/gtkentrycompletion.c b/gtk/gtkentrycompletion.c
index 439d856..d7ab7a2 100644
--- a/gtk/gtkentrycompletion.c
+++ b/gtk/gtkentrycompletion.c
@@ -339,7 +339,7 @@ gtk_entry_completion_class_init (GtkEntryCompletionClass *klass)
                   G_SIGNAL_RUN_LAST,
                   G_STRUCT_OFFSET (GtkEntryCompletionClass, action_activated),
                   NULL, NULL,
-                  _gtk_marshal_VOID__INT,
+                  NULL,
                   G_TYPE_NONE, 1,
                   G_TYPE_INT);
 
diff --git a/gtk/gtkexpander.c b/gtk/gtkexpander.c
index eb2ce2d..ad84d5f 100644
--- a/gtk/gtkexpander.c
+++ b/gtk/gtkexpander.c
@@ -412,7 +412,7 @@ gtk_expander_class_init (GtkExpanderClass *klass)
                   G_SIGNAL_RUN_LAST | G_SIGNAL_ACTION,
                   G_STRUCT_OFFSET (GtkExpanderClass, activate),
                   NULL, NULL,
-                  _gtk_marshal_VOID__VOID,
+                  NULL,
                   G_TYPE_NONE, 0);
 
   gtk_widget_class_set_accessible_type (widget_class, GTK_TYPE_EXPANDER_ACCESSIBLE);
diff --git a/gtk/gtkfilechooserbutton.c b/gtk/gtkfilechooserbutton.c
index 957040d..981b4d4 100644
--- a/gtk/gtkfilechooserbutton.c
+++ b/gtk/gtkfilechooserbutton.c
@@ -408,7 +408,7 @@ gtk_file_chooser_button_class_init (GtkFileChooserButtonClass * class)
 		  G_SIGNAL_RUN_FIRST,
 		  G_STRUCT_OFFSET (GtkFileChooserButtonClass, file_set),
 		  NULL, NULL,
-		  _gtk_marshal_VOID__VOID,
+		  NULL,
 		  G_TYPE_NONE, 0);
 
   /**
diff --git a/gtk/gtkfilechooserembed.c b/gtk/gtkfilechooserembed.c
index 336c2b3..7c64f37 100644
--- a/gtk/gtkfilechooserembed.c
+++ b/gtk/gtkfilechooserembed.c
@@ -154,14 +154,14 @@ gtk_file_chooser_embed_class_init (gpointer g_iface)
 		G_SIGNAL_RUN_LAST,
 		G_STRUCT_OFFSET (GtkFileChooserEmbedIface, default_size_changed),
 		NULL, NULL,
-		_gtk_marshal_VOID__VOID,
+		NULL,
 		G_TYPE_NONE, 0);
   g_signal_new (I_("response-requested"),
 		iface_type,
 		G_SIGNAL_RUN_LAST,
 		G_STRUCT_OFFSET (GtkFileChooserEmbedIface, response_requested),
 		NULL, NULL,
-		_gtk_marshal_VOID__VOID,
+		NULL,
 		G_TYPE_NONE, 0);
 }
 
diff --git a/gtk/gtkfilechooserentry.c b/gtk/gtkfilechooserentry.c
index 86f143e..0805b13 100644
--- a/gtk/gtkfilechooserentry.c
+++ b/gtk/gtkfilechooserentry.c
@@ -175,7 +175,7 @@ _gtk_file_chooser_entry_class_init (GtkFileChooserEntryClass *class)
                   G_SIGNAL_RUN_LAST | G_SIGNAL_ACTION,
                   0,
                   NULL, NULL,
-                  _gtk_marshal_VOID__VOID,
+                  NULL,
                   G_TYPE_NONE, 0);
 }
 
diff --git a/gtk/gtkfilechooserwidget.c b/gtk/gtkfilechooserwidget.c
index 87ab075..be9d856 100644
--- a/gtk/gtkfilechooserwidget.c
+++ b/gtk/gtkfilechooserwidget.c
@@ -8150,7 +8150,7 @@ gtk_file_chooser_widget_class_init (GtkFileChooserWidgetClass *class)
                                 G_SIGNAL_RUN_FIRST | G_SIGNAL_ACTION,
                                 G_CALLBACK (location_popup_handler),
                                 NULL, NULL,
-                                _gtk_marshal_VOID__STRING,
+                                NULL,
                                 G_TYPE_NONE, 1, G_TYPE_STRING);
 
   /**
@@ -8171,7 +8171,7 @@ gtk_file_chooser_widget_class_init (GtkFileChooserWidgetClass *class)
                                 G_SIGNAL_RUN_FIRST | G_SIGNAL_ACTION,
                                 G_CALLBACK (location_popup_on_paste_handler),
                                 NULL, NULL,
-                                _gtk_marshal_VOID__VOID,
+                                NULL,
                                 G_TYPE_NONE, 0);
 
   /**
@@ -8192,7 +8192,7 @@ gtk_file_chooser_widget_class_init (GtkFileChooserWidgetClass *class)
                                 G_SIGNAL_RUN_FIRST | G_SIGNAL_ACTION,
                                 G_CALLBACK (location_toggle_popup_handler),
                                 NULL, NULL,
-                                _gtk_marshal_VOID__VOID,
+                                NULL,
                                 G_TYPE_NONE, 0);
 
   /**
@@ -8213,7 +8213,7 @@ gtk_file_chooser_widget_class_init (GtkFileChooserWidgetClass *class)
                                 G_SIGNAL_RUN_FIRST | G_SIGNAL_ACTION,
                                 G_CALLBACK (up_folder_handler),
                                 NULL, NULL,
-                                _gtk_marshal_VOID__VOID,
+                                NULL,
                                 G_TYPE_NONE, 0);
 
   /**
@@ -8237,7 +8237,7 @@ gtk_file_chooser_widget_class_init (GtkFileChooserWidgetClass *class)
                                 G_SIGNAL_RUN_FIRST | G_SIGNAL_ACTION,
                                 G_CALLBACK (down_folder_handler),
                                 NULL, NULL,
-                                _gtk_marshal_VOID__VOID,
+                                NULL,
                                 G_TYPE_NONE, 0);
 
   /**
@@ -8258,7 +8258,7 @@ gtk_file_chooser_widget_class_init (GtkFileChooserWidgetClass *class)
                                 G_SIGNAL_RUN_FIRST | G_SIGNAL_ACTION,
                                 G_CALLBACK (home_folder_handler),
                                 NULL, NULL,
-                                _gtk_marshal_VOID__VOID,
+                                NULL,
                                 G_TYPE_NONE, 0);
 
   /**
@@ -8279,7 +8279,7 @@ gtk_file_chooser_widget_class_init (GtkFileChooserWidgetClass *class)
                                 G_SIGNAL_RUN_FIRST | G_SIGNAL_ACTION,
                                 G_CALLBACK (desktop_folder_handler),
                                 NULL, NULL,
-                                _gtk_marshal_VOID__VOID,
+                                NULL,
                                 G_TYPE_NONE, 0);
 
   /**
@@ -8306,7 +8306,7 @@ gtk_file_chooser_widget_class_init (GtkFileChooserWidgetClass *class)
                                 G_SIGNAL_RUN_FIRST | G_SIGNAL_ACTION,
                                 G_CALLBACK (quick_bookmark_handler),
                                 NULL, NULL,
-                                _gtk_marshal_VOID__INT,
+                                NULL,
                                 G_TYPE_NONE, 1, G_TYPE_INT);
 
   /**
@@ -8326,7 +8326,7 @@ gtk_file_chooser_widget_class_init (GtkFileChooserWidgetClass *class)
                                 G_SIGNAL_RUN_FIRST | G_SIGNAL_ACTION,
                                 G_CALLBACK (show_hidden_handler),
                                 NULL, NULL,
-                                _gtk_marshal_VOID__VOID,
+                                NULL,
                                 G_TYPE_NONE, 0);
 
   /**
@@ -8346,7 +8346,7 @@ gtk_file_chooser_widget_class_init (GtkFileChooserWidgetClass *class)
                                 G_SIGNAL_RUN_FIRST | G_SIGNAL_ACTION,
                                 G_CALLBACK (search_shortcut_handler),
                                 NULL, NULL,
-                                _gtk_marshal_VOID__VOID,
+                                NULL,
                                 G_TYPE_NONE, 0);
 
   /**
@@ -8366,7 +8366,7 @@ gtk_file_chooser_widget_class_init (GtkFileChooserWidgetClass *class)
                                 G_SIGNAL_RUN_FIRST | G_SIGNAL_ACTION,
                                 G_CALLBACK (recent_shortcut_handler),
                                 NULL, NULL,
-                                _gtk_marshal_VOID__VOID,
+                                NULL,
                                 G_TYPE_NONE, 0);
 
   /**
@@ -8386,7 +8386,7 @@ gtk_file_chooser_widget_class_init (GtkFileChooserWidgetClass *class)
                                 G_SIGNAL_RUN_FIRST | G_SIGNAL_ACTION,
                                 G_CALLBACK (places_shortcut_handler),
                                 NULL, NULL,
-                                _gtk_marshal_VOID__VOID,
+                                NULL,
                                 G_TYPE_NONE, 0);
 
   binding_set = gtk_binding_set_by_class (class);
diff --git a/gtk/gtkfilesystemmodel.c b/gtk/gtkfilesystemmodel.c
index 32e2cac..33714a8 100644
--- a/gtk/gtkfilesystemmodel.c
+++ b/gtk/gtkfilesystemmodel.c
@@ -1092,7 +1092,7 @@ _gtk_file_system_model_class_init (GtkFileSystemModelClass *class)
 		  G_SIGNAL_RUN_LAST,
 		  G_STRUCT_OFFSET (GtkFileSystemModelClass, finished_loading),
 		  NULL, NULL,
-		  _gtk_marshal_VOID__POINTER,
+		  NULL,
 		  G_TYPE_NONE, 1, G_TYPE_POINTER);
 }
 
diff --git a/gtk/gtkimcontext.c b/gtk/gtkimcontext.c
index 64f0f69..5ca199f 100644
--- a/gtk/gtkimcontext.c
+++ b/gtk/gtkimcontext.c
@@ -231,7 +231,7 @@ gtk_im_context_class_init (GtkIMContextClass *klass)
 		  G_SIGNAL_RUN_LAST,
 		  G_STRUCT_OFFSET (GtkIMContextClass, preedit_start),
 		  NULL, NULL,
-		  _gtk_marshal_VOID__VOID,
+		  NULL,
 		  G_TYPE_NONE, 0);
   /**
    * GtkIMContext::preedit-end:
@@ -246,7 +246,7 @@ gtk_im_context_class_init (GtkIMContextClass *klass)
 		  G_SIGNAL_RUN_LAST,
 		  G_STRUCT_OFFSET (GtkIMContextClass, preedit_end),
 		  NULL, NULL,
-		  _gtk_marshal_VOID__VOID,
+		  NULL,
 		  G_TYPE_NONE, 0);
   /**
    * GtkIMContext::preedit-changed:
@@ -263,7 +263,7 @@ gtk_im_context_class_init (GtkIMContextClass *klass)
 		  G_SIGNAL_RUN_LAST,
 		  G_STRUCT_OFFSET (GtkIMContextClass, preedit_changed),
 		  NULL, NULL,
-		  _gtk_marshal_VOID__VOID,
+		  NULL,
 		  G_TYPE_NONE, 0);
   /**
    * GtkIMContext::commit:
@@ -280,7 +280,7 @@ gtk_im_context_class_init (GtkIMContextClass *klass)
 		  G_SIGNAL_RUN_LAST,
 		  G_STRUCT_OFFSET (GtkIMContextClass, commit),
 		  NULL, NULL,
-		  _gtk_marshal_VOID__STRING,
+		  NULL,
 		  G_TYPE_NONE, 1,
 		  G_TYPE_STRING);
   /**
diff --git a/gtk/gtklabel.c b/gtk/gtklabel.c
index 3f382e0..a8925ff 100644
--- a/gtk/gtklabel.c
+++ b/gtk/gtklabel.c
@@ -711,7 +711,7 @@ gtk_label_class_init (GtkLabelClass *class)
 		  G_SIGNAL_RUN_LAST | G_SIGNAL_ACTION,
 		  G_STRUCT_OFFSET (GtkLabelClass, copy_clipboard),
 		  NULL, NULL,
-		  _gtk_marshal_VOID__VOID,
+		  NULL,
 		  G_TYPE_NONE, 0);
   
   /**
@@ -732,7 +732,7 @@ gtk_label_class_init (GtkLabelClass *class)
 		  G_SIGNAL_RUN_LAST,
 		  G_STRUCT_OFFSET (GtkLabelClass, populate_popup),
 		  NULL, NULL,
-		  _gtk_marshal_VOID__OBJECT,
+		  NULL,
 		  G_TYPE_NONE, 1,
 		  GTK_TYPE_MENU);
 
@@ -756,7 +756,7 @@ gtk_label_class_init (GtkLabelClass *class)
                                   G_SIGNAL_RUN_LAST | G_SIGNAL_ACTION,
                                   G_CALLBACK (gtk_label_activate_current_link),
                                   NULL, NULL,
-                                  _gtk_marshal_VOID__VOID,
+                                  NULL,
                                   G_TYPE_NONE, 0);
 
     /**
diff --git a/gtk/gtklevelbar.c b/gtk/gtklevelbar.c
index d293661..1081a82 100644
--- a/gtk/gtklevelbar.c
+++ b/gtk/gtklevelbar.c
@@ -1003,7 +1003,7 @@ gtk_level_bar_class_init (GtkLevelBarClass *klass)
                   G_SIGNAL_RUN_FIRST | G_SIGNAL_DETAILED,
                   G_STRUCT_OFFSET (GtkLevelBarClass, offset_changed),
                   NULL, NULL,
-                  _gtk_marshal_VOID__STRING,
+                  NULL,
                   G_TYPE_NONE,
                   1, G_TYPE_STRING);
 
diff --git a/gtk/gtklistbox.c b/gtk/gtklistbox.c
index 17f1907..57d1f33 100644
--- a/gtk/gtklistbox.c
+++ b/gtk/gtklistbox.c
@@ -3721,7 +3721,7 @@ gtk_list_box_row_class_init (GtkListBoxRowClass *klass)
                   G_SIGNAL_RUN_FIRST | G_SIGNAL_ACTION,
                   G_STRUCT_OFFSET (GtkListBoxRowClass, activate),
                   NULL, NULL,
-                  _gtk_marshal_VOID__VOID,
+                  NULL,
                   G_TYPE_NONE, 0);
 
   widget_class->activate_signal = row_signals[ROW__ACTIVATE];
diff --git a/gtk/gtkmarshalers.list b/gtk/gtkmarshalers.list
index 32c0ded..cb6f4fa 100644
--- a/gtk/gtkmarshalers.list
+++ b/gtk/gtkmarshalers.list
@@ -51,11 +51,8 @@ INT:POINTER
 OBJECT:VOID
 STRING:DOUBLE
 STRING:STRING
-VOID:DOUBLE
 VOID:DOUBLE,DOUBLE
-VOID:BOOLEAN
 VOID:BOOLEAN,BOOLEAN,BOOLEAN
-VOID:BOXED
 VOID:BOXED,BOXED
 VOID:BOXED,BOXED,POINTER
 VOID:BOXED,OBJECT
@@ -63,7 +60,6 @@ VOID:BOXED,STRING,INT
 VOID:BOXED,UINT
 VOID:BOXED,UINT,FLAGS
 VOID:BOXED,UINT,UINT
-VOID:ENUM
 VOID:ENUM,BOOLEAN
 VOID:ENUM,ENUM
 VOID:ENUM,FLOAT
@@ -73,13 +69,10 @@ VOID:ENUM,INT,BOOLEAN
 VOID:ENUM,INT,INT
 VOID:ENUM,BOXED
 VOID:ENUM,STRING
-VOID:FLAGS
-VOID:INT
 VOID:INT,BOOLEAN
 VOID:INT,INT
 VOID:INT,INT,BOXED
 VOID:INT,INT,INT
-VOID:OBJECT
 VOID:OBJECT,BOOLEAN
 VOID:OBJECT,BOXED,BOXED
 VOID:OBJECT,BOXED,UINT,UINT
@@ -102,13 +95,11 @@ VOID:OBJECT,OBJECT,STRING
 VOID:OBJECT,OBJECT,OBJECT
 VOID:OBJECT,OBJECT,BOXED,STRING
 VOID:OBJECT,OBJECT,POINTER,POINTER
-VOID:POINTER
 VOID:POINTER,INT
 VOID:POINTER,BOOLEAN
 VOID:POINTER,POINTER,BOOLEAN
 VOID:POINTER,POINTER,POINTER
 VOID:POINTER,UINT
-VOID:STRING
 VOID:STRING,BOXED
 VOID:STRING,STRING
 VOID:STRING,STRING,STRING
@@ -123,7 +114,6 @@ VOID:UINT,BOXED,UINT,FLAGS,FLAGS
 VOID:UINT,OBJECT,UINT,FLAGS,FLAGS
 VOID:UINT,STRING,UINT
 VOID:UINT,UINT
-VOID:VOID
 OBJECT:OBJECT,INT,INT
 VOID:OBJECT,STRING,POINTER,POINTER
 INT:INT
diff --git a/gtk/gtkmenu.c b/gtk/gtkmenu.c
index aa11036..8636bab 100644
--- a/gtk/gtkmenu.c
+++ b/gtk/gtkmenu.c
@@ -563,7 +563,7 @@ gtk_menu_class_init (GtkMenuClass *class)
                                 G_SIGNAL_RUN_LAST | G_SIGNAL_ACTION,
                                 G_CALLBACK (gtk_menu_real_move_scroll),
                                 NULL, NULL,
-                                _gtk_marshal_VOID__ENUM,
+                                NULL,
                                 G_TYPE_NONE, 1,
                                 GTK_TYPE_SCROLL_TYPE);
 
diff --git a/gtk/gtkmenuitem.c b/gtk/gtkmenuitem.c
index 17a71d4..09f7d54 100644
--- a/gtk/gtkmenuitem.c
+++ b/gtk/gtkmenuitem.c
@@ -671,7 +671,7 @@ gtk_menu_item_class_init (GtkMenuItemClass *klass)
                   G_SIGNAL_RUN_FIRST | G_SIGNAL_ACTION,
                   G_STRUCT_OFFSET (GtkMenuItemClass, activate),
                   NULL, NULL,
-                  _gtk_marshal_VOID__VOID,
+                  NULL,
                   G_TYPE_NONE, 0);
   widget_class->activate_signal = menu_item_signals[ACTIVATE];
 
@@ -689,7 +689,7 @@ gtk_menu_item_class_init (GtkMenuItemClass *klass)
                   G_SIGNAL_RUN_FIRST,
                   G_STRUCT_OFFSET (GtkMenuItemClass, activate_item),
                   NULL, NULL,
-                  _gtk_marshal_VOID__VOID,
+                  NULL,
                   G_TYPE_NONE, 0);
 
   menu_item_signals[TOGGLE_SIZE_REQUEST] =
@@ -698,7 +698,7 @@ gtk_menu_item_class_init (GtkMenuItemClass *klass)
                   G_SIGNAL_RUN_FIRST,
                   G_STRUCT_OFFSET (GtkMenuItemClass, toggle_size_request),
                   NULL, NULL,
-                  _gtk_marshal_VOID__POINTER,
+                  NULL,
                   G_TYPE_NONE, 1,
                   G_TYPE_POINTER);
 
@@ -708,7 +708,7 @@ gtk_menu_item_class_init (GtkMenuItemClass *klass)
                   G_SIGNAL_RUN_FIRST,
                   G_STRUCT_OFFSET (GtkMenuItemClass, toggle_size_allocate),
                   NULL, NULL,
-                  _gtk_marshal_VOID__INT,
+                  NULL,
                   G_TYPE_NONE, 1,
                   G_TYPE_INT);
 
@@ -718,7 +718,7 @@ gtk_menu_item_class_init (GtkMenuItemClass *klass)
                   G_SIGNAL_RUN_FIRST,
                   G_STRUCT_OFFSET (GtkMenuItemClass, select),
                   NULL, NULL,
-                  _gtk_marshal_VOID__VOID,
+                  NULL,
                   G_TYPE_NONE, 0);
 
   menu_item_signals[DESELECT] =
@@ -727,7 +727,7 @@ gtk_menu_item_class_init (GtkMenuItemClass *klass)
                   G_SIGNAL_RUN_FIRST,
                   G_STRUCT_OFFSET (GtkMenuItemClass, deselect),
                   NULL, NULL,
-                  _gtk_marshal_VOID__VOID,
+                  NULL,
                   G_TYPE_NONE, 0);
 
   /**
diff --git a/gtk/gtkmenushell.c b/gtk/gtkmenushell.c
index 22b917a..a68389e 100644
--- a/gtk/gtkmenushell.c
+++ b/gtk/gtkmenushell.c
@@ -227,7 +227,7 @@ gtk_menu_shell_class_init (GtkMenuShellClass *klass)
                   G_SIGNAL_RUN_FIRST,
                   G_STRUCT_OFFSET (GtkMenuShellClass, deactivate),
                   NULL, NULL,
-                  _gtk_marshal_VOID__VOID,
+                  NULL,
                   G_TYPE_NONE, 0);
 
   /**
@@ -243,7 +243,7 @@ gtk_menu_shell_class_init (GtkMenuShellClass *klass)
                   G_SIGNAL_RUN_FIRST,
                   G_STRUCT_OFFSET (GtkMenuShellClass, selection_done),
                   NULL, NULL,
-                  _gtk_marshal_VOID__VOID,
+                  NULL,
                   G_TYPE_NONE, 0);
 
   /**
@@ -260,7 +260,7 @@ gtk_menu_shell_class_init (GtkMenuShellClass *klass)
                   G_SIGNAL_RUN_LAST | G_SIGNAL_ACTION,
                   G_STRUCT_OFFSET (GtkMenuShellClass, move_current),
                   NULL, NULL,
-                  _gtk_marshal_VOID__ENUM,
+                  NULL,
                   G_TYPE_NONE, 1,
                   GTK_TYPE_MENU_DIRECTION_TYPE);
 
@@ -278,7 +278,7 @@ gtk_menu_shell_class_init (GtkMenuShellClass *klass)
                   G_SIGNAL_RUN_LAST | G_SIGNAL_ACTION,
                   G_STRUCT_OFFSET (GtkMenuShellClass, activate_current),
                   NULL, NULL,
-                  _gtk_marshal_VOID__BOOLEAN,
+                  NULL,
                   G_TYPE_NONE, 1,
                   G_TYPE_BOOLEAN);
 
@@ -295,7 +295,7 @@ gtk_menu_shell_class_init (GtkMenuShellClass *klass)
                   G_SIGNAL_RUN_LAST | G_SIGNAL_ACTION,
                   G_STRUCT_OFFSET (GtkMenuShellClass, cancel),
                   NULL, NULL,
-                  _gtk_marshal_VOID__VOID,
+                  NULL,
                   G_TYPE_NONE, 0);
 
   /**
@@ -312,7 +312,7 @@ gtk_menu_shell_class_init (GtkMenuShellClass *klass)
                                 G_SIGNAL_RUN_LAST | G_SIGNAL_ACTION,
                                 G_CALLBACK (gtk_real_menu_shell_cycle_focus),
                                 NULL, NULL,
-                                _gtk_marshal_VOID__ENUM,
+                                NULL,
                                 G_TYPE_NONE, 1,
                                 GTK_TYPE_DIRECTION_TYPE);
 
diff --git a/gtk/gtknotebook.c b/gtk/gtknotebook.c
index 60f4b15..29e5b31 100644
--- a/gtk/gtknotebook.c
+++ b/gtk/gtknotebook.c
@@ -1080,7 +1080,7 @@ gtk_notebook_class_init (GtkNotebookClass *class)
                   G_SIGNAL_RUN_LAST | G_SIGNAL_ACTION,
                   G_STRUCT_OFFSET (GtkNotebookClass, move_focus_out),
                   NULL, NULL,
-                  _gtk_marshal_VOID__ENUM,
+                  NULL,
                   G_TYPE_NONE, 1,
                   GTK_TYPE_DIRECTION_TYPE);
   notebook_signals[REORDER_TAB] =
diff --git a/gtk/gtkplacessidebar.c b/gtk/gtkplacessidebar.c
index 3f1f810..b75c30f 100644
--- a/gtk/gtkplacessidebar.c
+++ b/gtk/gtkplacessidebar.c
@@ -4237,7 +4237,7 @@ gtk_places_sidebar_class_init (GtkPlacesSidebarClass *class)
                         G_SIGNAL_RUN_FIRST,
                         G_STRUCT_OFFSET (GtkPlacesSidebarClass, show_connect_to_server),
                         NULL, NULL,
-                        _gtk_marshal_VOID__VOID,
+                        NULL,
                         G_TYPE_NONE, 0);
 
   /**
@@ -4257,7 +4257,7 @@ gtk_places_sidebar_class_init (GtkPlacesSidebarClass *class)
                         G_SIGNAL_RUN_FIRST,
                         G_STRUCT_OFFSET (GtkPlacesSidebarClass, show_enter_location),
                         NULL, NULL,
-                        _gtk_marshal_VOID__VOID,
+                        NULL,
                         G_TYPE_NONE, 0);
 
   /**
@@ -4369,7 +4369,7 @@ gtk_places_sidebar_class_init (GtkPlacesSidebarClass *class)
                         G_SIGNAL_RUN_FIRST | G_SIGNAL_DEPRECATED,
                         G_STRUCT_OFFSET (GtkPlacesSidebarClass, show_other_locations),
                         NULL, NULL,
-                        _gtk_marshal_VOID__VOID,
+                        NULL,
                         G_TYPE_NONE, 0);
 
   /**
@@ -4391,7 +4391,7 @@ gtk_places_sidebar_class_init (GtkPlacesSidebarClass *class)
                         G_SIGNAL_RUN_FIRST,
                         G_STRUCT_OFFSET (GtkPlacesSidebarClass, show_other_locations_with_flags),
                         NULL, NULL,
-                        _gtk_marshal_VOID__VOID,
+                        NULL,
                         G_TYPE_NONE, 1,
                         GTK_TYPE_PLACES_OPEN_FLAGS);
 
@@ -4413,7 +4413,7 @@ gtk_places_sidebar_class_init (GtkPlacesSidebarClass *class)
                         G_SIGNAL_RUN_FIRST,
                         G_STRUCT_OFFSET (GtkPlacesSidebarClass, mount),
                         NULL, NULL,
-                        _gtk_marshal_VOID__OBJECT,
+                        NULL,
                         G_TYPE_NONE,
                         1,
                         G_TYPE_MOUNT_OPERATION);
@@ -4435,7 +4435,7 @@ gtk_places_sidebar_class_init (GtkPlacesSidebarClass *class)
                         G_SIGNAL_RUN_FIRST,
                         G_STRUCT_OFFSET (GtkPlacesSidebarClass, unmount),
                         NULL, NULL,
-                        _gtk_marshal_VOID__OBJECT,
+                        NULL,
                         G_TYPE_NONE,
                         1,
                         G_TYPE_MOUNT_OPERATION);
diff --git a/gtk/gtkplug.c b/gtk/gtkplug.c
index 1e6942d..4e72543 100644
--- a/gtk/gtkplug.c
+++ b/gtk/gtkplug.c
@@ -225,7 +225,7 @@ gtk_plug_class_init (GtkPlugClass *class)
 		  G_SIGNAL_RUN_LAST,
 		  G_STRUCT_OFFSET (GtkPlugClass, embedded),
 		  NULL, NULL,
-		  _gtk_marshal_VOID__VOID,
+		  NULL,
 		  G_TYPE_NONE, 0);
 }
 
diff --git a/gtk/gtkradiobutton.c b/gtk/gtkradiobutton.c
index 59ca75c..d38fe95 100644
--- a/gtk/gtkradiobutton.c
+++ b/gtk/gtkradiobutton.c
@@ -213,7 +213,7 @@ gtk_radio_button_class_init (GtkRadioButtonClass *class)
 				       G_SIGNAL_RUN_FIRST,
 				       G_STRUCT_OFFSET (GtkRadioButtonClass, group_changed),
 				       NULL, NULL,
-				       _gtk_marshal_VOID__VOID,
+				       NULL,
 				       G_TYPE_NONE, 0);
 
   gtk_widget_class_set_accessible_type (widget_class, GTK_TYPE_RADIO_BUTTON_ACCESSIBLE);
diff --git a/gtk/gtkradiomenuitem.c b/gtk/gtkradiomenuitem.c
index b8bd75d..8e66f85 100644
--- a/gtk/gtkradiomenuitem.c
+++ b/gtk/gtkradiomenuitem.c
@@ -444,7 +444,7 @@ gtk_radio_menu_item_class_init (GtkRadioMenuItemClass *klass)
 				       G_SIGNAL_RUN_FIRST,
 				       G_STRUCT_OFFSET (GtkRadioMenuItemClass, group_changed),
 				       NULL, NULL,
-				       _gtk_marshal_VOID__VOID,
+				       NULL,
 				       G_TYPE_NONE, 0);
 }
 
diff --git a/gtk/gtkrange.c b/gtk/gtkrange.c
index c124ca2..282110f 100644
--- a/gtk/gtkrange.c
+++ b/gtk/gtkrange.c
@@ -354,7 +354,7 @@ gtk_range_class_init (GtkRangeClass *class)
                   G_SIGNAL_RUN_LAST,
                   G_STRUCT_OFFSET (GtkRangeClass, value_changed),
                   NULL, NULL,
-                  _gtk_marshal_VOID__VOID,
+                  NULL,
                   G_TYPE_NONE, 0);
 
   /**
@@ -371,7 +371,7 @@ gtk_range_class_init (GtkRangeClass *class)
                   G_SIGNAL_RUN_LAST,
                   G_STRUCT_OFFSET (GtkRangeClass, adjust_bounds),
                   NULL, NULL,
-                  _gtk_marshal_VOID__DOUBLE,
+                  NULL,
                   G_TYPE_NONE, 1,
                   G_TYPE_DOUBLE);
 
@@ -388,7 +388,7 @@ gtk_range_class_init (GtkRangeClass *class)
                   G_SIGNAL_RUN_LAST | G_SIGNAL_ACTION,
                   G_STRUCT_OFFSET (GtkRangeClass, move_slider),
                   NULL, NULL,
-                  _gtk_marshal_VOID__ENUM,
+                  NULL,
                   G_TYPE_NONE, 1,
                   GTK_TYPE_SCROLL_TYPE);
 
diff --git a/gtk/gtkscalebutton.c b/gtk/gtkscalebutton.c
index 78204e2..81a689e 100644
--- a/gtk/gtkscalebutton.c
+++ b/gtk/gtkscalebutton.c
@@ -262,7 +262,7 @@ gtk_scale_button_class_init (GtkScaleButtonClass *klass)
 		  G_SIGNAL_RUN_LAST,
 		  G_STRUCT_OFFSET (GtkScaleButtonClass, value_changed),
 		  NULL, NULL,
-		  _gtk_marshal_VOID__DOUBLE,
+		  NULL,
 		  G_TYPE_NONE, 1, G_TYPE_DOUBLE);
 
   /**
diff --git a/gtk/gtkscrolledwindow.c b/gtk/gtkscrolledwindow.c
index 0a7a52f..082a0dc 100644
--- a/gtk/gtkscrolledwindow.c
+++ b/gtk/gtkscrolledwindow.c
@@ -785,7 +785,7 @@ gtk_scrolled_window_class_init (GtkScrolledWindowClass *class)
                   G_SIGNAL_RUN_LAST | G_SIGNAL_ACTION,
                   G_STRUCT_OFFSET (GtkScrolledWindowClass, move_focus_out),
                   NULL, NULL,
-                  _gtk_marshal_VOID__ENUM,
+                  NULL,
                   G_TYPE_NONE, 1,
                   GTK_TYPE_DIRECTION_TYPE);
 
diff --git a/gtk/gtksearchentry.c b/gtk/gtksearchentry.c
index 64c31a2..ef725ea 100644
--- a/gtk/gtksearchentry.c
+++ b/gtk/gtksearchentry.c
@@ -178,7 +178,7 @@ gtk_search_entry_class_init (GtkSearchEntryClass *klass)
                   G_SIGNAL_RUN_LAST,
                   G_STRUCT_OFFSET (GtkSearchEntryClass, search_changed),
                   NULL, NULL,
-                  _gtk_marshal_VOID__VOID,
+                  NULL,
                   G_TYPE_NONE, 0);
 
   /**
@@ -202,7 +202,7 @@ gtk_search_entry_class_init (GtkSearchEntryClass *klass)
                   G_SIGNAL_RUN_LAST | G_SIGNAL_ACTION,
                   G_STRUCT_OFFSET (GtkSearchEntryClass, next_match),
                   NULL, NULL,
-                  _gtk_marshal_VOID__VOID,
+                  NULL,
                   G_TYPE_NONE, 0);
 
   /**
@@ -226,7 +226,7 @@ gtk_search_entry_class_init (GtkSearchEntryClass *klass)
                   G_SIGNAL_RUN_LAST | G_SIGNAL_ACTION,
                   G_STRUCT_OFFSET (GtkSearchEntryClass, previous_match),
                   NULL, NULL,
-                  _gtk_marshal_VOID__VOID,
+                  NULL,
                   G_TYPE_NONE, 0);
 
   /**
@@ -249,7 +249,7 @@ gtk_search_entry_class_init (GtkSearchEntryClass *klass)
                   G_SIGNAL_RUN_LAST | G_SIGNAL_ACTION,
                   G_STRUCT_OFFSET (GtkSearchEntryClass, stop_search),
                   NULL, NULL,
-                  _gtk_marshal_VOID__VOID,
+                  NULL,
                   G_TYPE_NONE, 0);
 
   binding_set = gtk_binding_set_by_class (klass);
diff --git a/gtk/gtksocket.c b/gtk/gtksocket.c
index 876d5aa..1e19d0b 100644
--- a/gtk/gtksocket.c
+++ b/gtk/gtksocket.c
@@ -243,7 +243,7 @@ gtk_socket_class_init (GtkSocketClass *class)
 		  G_SIGNAL_RUN_LAST,
 		  G_STRUCT_OFFSET (GtkSocketClass, plug_added),
 		  NULL, NULL,
-		  _gtk_marshal_VOID__VOID,
+		  NULL,
 		  G_TYPE_NONE, 0);
 
   /**
diff --git a/gtk/gtkspinbutton.c b/gtk/gtkspinbutton.c
index 7ddaac6..9e5e4c7 100644
--- a/gtk/gtkspinbutton.c
+++ b/gtk/gtkspinbutton.c
@@ -530,7 +530,7 @@ gtk_spin_button_class_init (GtkSpinButtonClass *class)
                   G_SIGNAL_RUN_LAST,
                   G_STRUCT_OFFSET (GtkSpinButtonClass, value_changed),
                   NULL, NULL,
-                  _gtk_marshal_VOID__VOID,
+                  NULL,
                   G_TYPE_NONE, 0);
 
   /**
@@ -548,7 +548,7 @@ gtk_spin_button_class_init (GtkSpinButtonClass *class)
                   G_SIGNAL_RUN_LAST,
                   G_STRUCT_OFFSET (GtkSpinButtonClass, wrapped),
                   NULL, NULL,
-                  _gtk_marshal_VOID__VOID,
+                  NULL,
                   G_TYPE_NONE, 0);
 
   /* Action signals */
@@ -572,7 +572,7 @@ gtk_spin_button_class_init (GtkSpinButtonClass *class)
                   G_SIGNAL_RUN_LAST | G_SIGNAL_ACTION,
                   G_STRUCT_OFFSET (GtkSpinButtonClass, change_value),
                   NULL, NULL,
-                  _gtk_marshal_VOID__ENUM,
+                  NULL,
                   G_TYPE_NONE, 1,
                   GTK_TYPE_SCROLL_TYPE);
 
diff --git a/gtk/gtkswitch.c b/gtk/gtkswitch.c
index b09b3b4..4b5e270 100644
--- a/gtk/gtkswitch.c
+++ b/gtk/gtkswitch.c
@@ -982,7 +982,7 @@ gtk_switch_class_init (GtkSwitchClass *klass)
                   G_SIGNAL_RUN_FIRST | G_SIGNAL_ACTION,
                   G_STRUCT_OFFSET (GtkSwitchClass, activate),
                   NULL, NULL,
-                  _gtk_marshal_VOID__VOID,
+                  NULL,
                   G_TYPE_NONE, 0);
   widget_class->activate_signal = signals[ACTIVATE];
 
diff --git a/gtk/gtktextbuffer.c b/gtk/gtktextbuffer.c
index ef17e92..644e2cf 100644
--- a/gtk/gtktextbuffer.c
+++ b/gtk/gtktextbuffer.c
@@ -427,7 +427,7 @@ gtk_text_buffer_class_init (GtkTextBufferClass *klass)
                   G_SIGNAL_RUN_LAST,                   
                   G_STRUCT_OFFSET (GtkTextBufferClass, changed),
                   NULL, NULL,
-                  _gtk_marshal_VOID__VOID,
+                  NULL,
                   G_TYPE_NONE,
                   0);
 
@@ -447,7 +447,7 @@ gtk_text_buffer_class_init (GtkTextBufferClass *klass)
                   G_SIGNAL_RUN_LAST,
                   G_STRUCT_OFFSET (GtkTextBufferClass, modified_changed),
                   NULL, NULL,
-                  _gtk_marshal_VOID__VOID,
+                  NULL,
                   G_TYPE_NONE,
                   0);
 
@@ -496,7 +496,7 @@ gtk_text_buffer_class_init (GtkTextBufferClass *klass)
                   G_SIGNAL_RUN_LAST,                   
                   G_STRUCT_OFFSET (GtkTextBufferClass, mark_deleted),
                   NULL, NULL,
-                  _gtk_marshal_VOID__OBJECT,
+                  NULL,
                   G_TYPE_NONE,
                   1,
                   GTK_TYPE_TEXT_MARK);
@@ -591,7 +591,7 @@ gtk_text_buffer_class_init (GtkTextBufferClass *klass)
                   G_SIGNAL_RUN_LAST,                   
                   G_STRUCT_OFFSET (GtkTextBufferClass, begin_user_action),
                   NULL, NULL,
-                  _gtk_marshal_VOID__VOID,
+                  NULL,
                   G_TYPE_NONE,
                   0);
 
@@ -617,7 +617,7 @@ gtk_text_buffer_class_init (GtkTextBufferClass *klass)
                   G_SIGNAL_RUN_LAST,                   
                   G_STRUCT_OFFSET (GtkTextBufferClass, end_user_action),
                   NULL, NULL,
-                  _gtk_marshal_VOID__VOID,
+                  NULL,
                   G_TYPE_NONE,
                   0);
 
@@ -638,7 +638,7 @@ gtk_text_buffer_class_init (GtkTextBufferClass *klass)
                   G_SIGNAL_RUN_LAST,
                   G_STRUCT_OFFSET (GtkTextBufferClass, paste_done),
                   NULL, NULL,
-                  _gtk_marshal_VOID__OBJECT,
+                  NULL,
                   G_TYPE_NONE,
                   1,
                   GTK_TYPE_CLIPBOARD);
diff --git a/gtk/gtktextlayout.c b/gtk/gtktextlayout.c
index 6829d45..9bd5582 100644
--- a/gtk/gtktextlayout.c
+++ b/gtk/gtktextlayout.c
@@ -242,7 +242,7 @@ gtk_text_layout_class_init (GtkTextLayoutClass *klass)
                   G_SIGNAL_RUN_LAST,
                   G_STRUCT_OFFSET (GtkTextLayoutClass, invalidated),
                   NULL, NULL,
-                  _gtk_marshal_VOID__VOID,
+                  NULL,
                   G_TYPE_NONE,
                   0);
 
diff --git a/gtk/gtktexttagtable.c b/gtk/gtktexttagtable.c
index 997ae95..7c4731c 100644
--- a/gtk/gtktexttagtable.c
+++ b/gtk/gtktexttagtable.c
@@ -128,7 +128,7 @@ gtk_text_tag_table_class_init (GtkTextTagTableClass *klass)
                   G_SIGNAL_RUN_LAST,
                   G_STRUCT_OFFSET (GtkTextTagTableClass, tag_added),
                   NULL, NULL,
-                  _gtk_marshal_VOID__OBJECT,
+                  NULL,
                   G_TYPE_NONE,
                   1,
                   GTK_TYPE_TEXT_TAG);
@@ -144,7 +144,7 @@ gtk_text_tag_table_class_init (GtkTextTagTableClass *klass)
                   G_SIGNAL_RUN_LAST,
                   G_STRUCT_OFFSET (GtkTextTagTableClass, tag_removed),
                   NULL, NULL,
-                  _gtk_marshal_VOID__OBJECT,
+                  NULL,
                   G_TYPE_NONE,
                   1,
                   GTK_TYPE_TEXT_TAG);
diff --git a/gtk/gtktextview.c b/gtk/gtktextview.c
index 403aad4..01b60e5 100644
--- a/gtk/gtktextview.c
+++ b/gtk/gtktextview.c
@@ -1165,7 +1165,7 @@ G_GNUC_END_IGNORE_DEPRECATIONS
 		  G_SIGNAL_RUN_LAST | G_SIGNAL_ACTION,
 		  G_STRUCT_OFFSET (GtkTextViewClass, set_anchor),
 		  NULL, NULL,
-		  _gtk_marshal_VOID__VOID,
+		  NULL,
 		  G_TYPE_NONE, 0);
 
   /**
@@ -1186,7 +1186,7 @@ G_GNUC_END_IGNORE_DEPRECATIONS
 		  G_SIGNAL_RUN_LAST | G_SIGNAL_ACTION,
 		  G_STRUCT_OFFSET (GtkTextViewClass, insert_at_cursor),
 		  NULL, NULL,
-		  _gtk_marshal_VOID__STRING,
+		  NULL,
 		  G_TYPE_NONE, 1,
 		  G_TYPE_STRING);
 
@@ -1237,7 +1237,7 @@ G_GNUC_END_IGNORE_DEPRECATIONS
 		  G_SIGNAL_RUN_LAST | G_SIGNAL_ACTION,
 		  G_STRUCT_OFFSET (GtkTextViewClass, backspace),
 		  NULL, NULL,
-		  _gtk_marshal_VOID__VOID,
+		  NULL,
 		  G_TYPE_NONE, 0);
 
   /**
@@ -1257,7 +1257,7 @@ G_GNUC_END_IGNORE_DEPRECATIONS
 		  G_SIGNAL_RUN_LAST | G_SIGNAL_ACTION,
 		  G_STRUCT_OFFSET (GtkTextViewClass, cut_clipboard),
 		  NULL, NULL,
-		  _gtk_marshal_VOID__VOID,
+		  NULL,
 		  G_TYPE_NONE, 0);
 
   /**
@@ -1277,7 +1277,7 @@ G_GNUC_END_IGNORE_DEPRECATIONS
 		  G_SIGNAL_RUN_LAST | G_SIGNAL_ACTION,
 		  G_STRUCT_OFFSET (GtkTextViewClass, copy_clipboard),
 		  NULL, NULL,
-		  _gtk_marshal_VOID__VOID,
+		  NULL,
 		  G_TYPE_NONE, 0);
 
   /**
@@ -1298,7 +1298,7 @@ G_GNUC_END_IGNORE_DEPRECATIONS
 		  G_SIGNAL_RUN_LAST | G_SIGNAL_ACTION,
 		  G_STRUCT_OFFSET (GtkTextViewClass, paste_clipboard),
 		  NULL, NULL,
-		  _gtk_marshal_VOID__VOID,
+		  NULL,
 		  G_TYPE_NONE, 0);
 
   /**
@@ -1317,7 +1317,7 @@ G_GNUC_END_IGNORE_DEPRECATIONS
 		  G_SIGNAL_RUN_LAST | G_SIGNAL_ACTION,
 		  G_STRUCT_OFFSET (GtkTextViewClass, toggle_overwrite),
 		  NULL, NULL,
-		  _gtk_marshal_VOID__VOID,
+		  NULL,
 		  G_TYPE_NONE, 0);
 
   /**
@@ -1346,7 +1346,7 @@ G_GNUC_END_IGNORE_DEPRECATIONS
 		  G_SIGNAL_RUN_LAST,
 		  G_STRUCT_OFFSET (GtkTextViewClass, populate_popup),
 		  NULL, NULL,
-		  _gtk_marshal_VOID__OBJECT,
+		  NULL,
 		  G_TYPE_NONE, 1,
 		  GTK_TYPE_WIDGET);
   
@@ -1369,7 +1369,7 @@ G_GNUC_END_IGNORE_DEPRECATIONS
                                 G_SIGNAL_RUN_LAST | G_SIGNAL_ACTION,
                                 G_CALLBACK (gtk_text_view_select_all),
                                 NULL, NULL,
-                                _gtk_marshal_VOID__BOOLEAN,
+                                NULL,
                                 G_TYPE_NONE, 1, G_TYPE_BOOLEAN);
 
   /**
@@ -1389,7 +1389,7 @@ G_GNUC_END_IGNORE_DEPRECATIONS
                                 G_SIGNAL_RUN_LAST | G_SIGNAL_ACTION,
                                 G_CALLBACK (gtk_text_view_toggle_cursor_visible),
                                 NULL, NULL,
-                                _gtk_marshal_VOID__VOID,
+                                NULL,
                                 G_TYPE_NONE, 0);
 
   /**
@@ -1412,7 +1412,7 @@ G_GNUC_END_IGNORE_DEPRECATIONS
                                 G_SIGNAL_RUN_LAST | G_SIGNAL_ACTION,
                                 NULL,
                                 NULL, NULL,
-                                _gtk_marshal_VOID__STRING,
+                                NULL,
                                 G_TYPE_NONE, 1,
                                 G_TYPE_STRING);
 
diff --git a/gtk/gtktogglebutton.c b/gtk/gtktogglebutton.c
index ac695d1..2421b2e 100644
--- a/gtk/gtktogglebutton.c
+++ b/gtk/gtktogglebutton.c
@@ -216,7 +216,7 @@ gtk_toggle_button_class_init (GtkToggleButtonClass *class)
 		  G_SIGNAL_RUN_FIRST,
 		  G_STRUCT_OFFSET (GtkToggleButtonClass, toggled),
 		  NULL, NULL,
-		  _gtk_marshal_VOID__VOID,
+		  NULL,
 		  G_TYPE_NONE, 0);
 
   gtk_widget_class_set_accessible_type (widget_class, GTK_TYPE_TOGGLE_BUTTON_ACCESSIBLE);
diff --git a/gtk/gtktoolitem.c b/gtk/gtktoolitem.c
index af1857d..c877985 100644
--- a/gtk/gtktoolitem.c
+++ b/gtk/gtktoolitem.c
@@ -254,7 +254,7 @@ gtk_tool_item_class_init (GtkToolItemClass *klass)
 		  G_SIGNAL_RUN_LAST,
 		  G_STRUCT_OFFSET (GtkToolItemClass, toolbar_reconfigured),
 		  NULL, NULL,
-		  _gtk_marshal_VOID__VOID,
+		  NULL,
 		  G_TYPE_NONE, 0);
 
   gtk_widget_class_set_css_name (widget_class, "toolitem");
diff --git a/gtk/gtktrashmonitor.c b/gtk/gtktrashmonitor.c
index 73d0590..8d9ef6f 100644
--- a/gtk/gtktrashmonitor.c
+++ b/gtk/gtktrashmonitor.c
@@ -88,7 +88,7 @@ _gtk_trash_monitor_class_init (GtkTrashMonitorClass *class)
 		  G_SIGNAL_RUN_FIRST,
 		  G_STRUCT_OFFSET (GtkTrashMonitorClass, trash_state_changed),
 		  NULL, NULL,
-		  _gtk_marshal_VOID__VOID,
+		  NULL,
 		  G_TYPE_NONE, 0);
 }
 
diff --git a/gtk/gtktreemenu.c b/gtk/gtktreemenu.c
index a703651..057bc96 100644
--- a/gtk/gtktreemenu.c
+++ b/gtk/gtktreemenu.c
@@ -249,7 +249,7 @@ _gtk_tree_menu_class_init (GtkTreeMenuClass *class)
                   G_SIGNAL_RUN_FIRST,
                   0, /* No class closure here */
                   NULL, NULL,
-                  _gtk_marshal_VOID__STRING,
+                  NULL,
                   G_TYPE_NONE, 1, G_TYPE_STRING);
 
   /*
diff --git a/gtk/gtktreemodel.c b/gtk/gtktreemodel.c
index 500658b..303a911 100644
--- a/gtk/gtktreemodel.c
+++ b/gtk/gtktreemodel.c
@@ -446,7 +446,7 @@ gtk_tree_model_base_init (gpointer g_class)
                        G_SIGNAL_RUN_FIRST,
                        closure,
                        NULL, NULL,
-                       _gtk_marshal_VOID__BOXED,
+                       NULL,
                        G_TYPE_NONE, 1,
                        row_deleted_params);
 
diff --git a/gtk/gtktreeselection.c b/gtk/gtktreeselection.c
index dc9ce5a..7150d1d 100644
--- a/gtk/gtktreeselection.c
+++ b/gtk/gtktreeselection.c
@@ -153,7 +153,7 @@ gtk_tree_selection_class_init (GtkTreeSelectionClass *class)
 		  G_SIGNAL_RUN_FIRST,
 		  G_STRUCT_OFFSET (GtkTreeSelectionClass, changed),
 		  NULL, NULL,
-		  _gtk_marshal_VOID__VOID,
+		  NULL,
 		  G_TYPE_NONE, 0);
 }
 
diff --git a/gtk/gtktreesortable.c b/gtk/gtktreesortable.c
index 1cb4274..4b3dac4 100644
--- a/gtk/gtktreesortable.c
+++ b/gtk/gtktreesortable.c
@@ -86,7 +86,7 @@ gtk_tree_sortable_base_init (gpointer g_class)
                     G_SIGNAL_RUN_LAST,
                     G_STRUCT_OFFSET (GtkTreeSortableIface, sort_column_changed),
                     NULL, NULL,
-                    _gtk_marshal_VOID__VOID,
+                    NULL,
                     G_TYPE_NONE, 0);
       initialized = TRUE;
     }
diff --git a/gtk/gtktreeview.c b/gtk/gtktreeview.c
index 935ee66..f182530 100644
--- a/gtk/gtktreeview.c
+++ b/gtk/gtktreeview.c
@@ -1439,7 +1439,7 @@ G_GNUC_END_IGNORE_DEPRECATIONS
 		  G_SIGNAL_RUN_LAST,
 		  G_STRUCT_OFFSET (GtkTreeViewClass, columns_changed),
 		  NULL, NULL,
-		  _gtk_marshal_VOID__VOID,
+		  NULL,
 		  G_TYPE_NONE, 0);
 
   /**
@@ -1454,7 +1454,7 @@ G_GNUC_END_IGNORE_DEPRECATIONS
 		  G_SIGNAL_RUN_LAST,
 		  G_STRUCT_OFFSET (GtkTreeViewClass, cursor_changed),
 		  NULL, NULL,
-		  _gtk_marshal_VOID__VOID,
+		  NULL,
 		  G_TYPE_NONE, 0);
 
   /**
diff --git a/gtk/gtktreeviewcolumn.c b/gtk/gtktreeviewcolumn.c
index 26f452a..055d938 100644
--- a/gtk/gtktreeviewcolumn.c
+++ b/gtk/gtktreeviewcolumn.c
@@ -238,7 +238,7 @@ gtk_tree_view_column_class_init (GtkTreeViewColumnClass *class)
                   G_SIGNAL_RUN_LAST,
                   G_STRUCT_OFFSET (GtkTreeViewColumnClass, clicked),
                   NULL, NULL,
-                  _gtk_marshal_VOID__VOID,
+                  NULL,
                   G_TYPE_NONE, 0);
 
   tree_column_props[PROP_VISIBLE] =
diff --git a/gtk/gtkwidget.c b/gtk/gtkwidget.c
index cbfb673..4c1d798 100644
--- a/gtk/gtkwidget.c
+++ b/gtk/gtkwidget.c
@@ -1642,7 +1642,7 @@ G_GNUC_END_IGNORE_DEPRECATIONS
                   G_SIGNAL_RUN_CLEANUP | G_SIGNAL_NO_RECURSE | G_SIGNAL_NO_HOOKS,
                   G_STRUCT_OFFSET (GtkWidgetClass, destroy),
                   NULL, NULL,
-                  _gtk_marshal_VOID__VOID,
+                  NULL,
                   G_TYPE_NONE, 0);
 
   /**
@@ -1658,7 +1658,7 @@ G_GNUC_END_IGNORE_DEPRECATIONS
 		  G_SIGNAL_RUN_FIRST,
 		  G_STRUCT_OFFSET (GtkWidgetClass, show),
 		  NULL, NULL,
-		  _gtk_marshal_VOID__VOID,
+		  NULL,
 		  G_TYPE_NONE, 0);
 
   /**
@@ -1674,7 +1674,7 @@ G_GNUC_END_IGNORE_DEPRECATIONS
 		  G_SIGNAL_RUN_FIRST,
 		  G_STRUCT_OFFSET (GtkWidgetClass, hide),
 		  NULL, NULL,
-		  _gtk_marshal_VOID__VOID,
+		  NULL,
 		  G_TYPE_NONE, 0);
 
   /**
@@ -1697,7 +1697,7 @@ G_GNUC_END_IGNORE_DEPRECATIONS
 		  G_SIGNAL_RUN_FIRST,
 		  G_STRUCT_OFFSET (GtkWidgetClass, map),
 		  NULL, NULL,
-		  _gtk_marshal_VOID__VOID,
+		  NULL,
 		  G_TYPE_NONE, 0);
 
   /**
@@ -1717,7 +1717,7 @@ G_GNUC_END_IGNORE_DEPRECATIONS
 		  G_SIGNAL_RUN_FIRST,
 		  G_STRUCT_OFFSET (GtkWidgetClass, unmap),
 		  NULL, NULL,
-		  _gtk_marshal_VOID__VOID,
+		  NULL,
 		  G_TYPE_NONE, 0);
 
   /**
@@ -1734,7 +1734,7 @@ G_GNUC_END_IGNORE_DEPRECATIONS
 		  G_SIGNAL_RUN_FIRST,
 		  G_STRUCT_OFFSET (GtkWidgetClass, realize),
 		  NULL, NULL,
-		  _gtk_marshal_VOID__VOID,
+		  NULL,
 		  G_TYPE_NONE, 0);
 
   /**
@@ -1752,7 +1752,7 @@ G_GNUC_END_IGNORE_DEPRECATIONS
 		  G_SIGNAL_RUN_LAST,
 		  G_STRUCT_OFFSET (GtkWidgetClass, unrealize),
 		  NULL, NULL,
-		  _gtk_marshal_VOID__VOID,
+		  NULL,
 		  G_TYPE_NONE, 0);
 
   /**
@@ -1767,7 +1767,7 @@ G_GNUC_END_IGNORE_DEPRECATIONS
 		  G_SIGNAL_RUN_FIRST,
 		  G_STRUCT_OFFSET (GtkWidgetClass, size_allocate),
 		  NULL, NULL,
-		  _gtk_marshal_VOID__BOXED,
+		  NULL,
 		  G_TYPE_NONE, 1,
 		  GDK_TYPE_RECTANGLE | G_SIGNAL_TYPE_STATIC_SCOPE);
 
@@ -1787,7 +1787,7 @@ G_GNUC_END_IGNORE_DEPRECATIONS
 		  G_SIGNAL_RUN_FIRST | G_SIGNAL_DEPRECATED,
 		  G_STRUCT_OFFSET (GtkWidgetClass, state_changed),
 		  NULL, NULL,
-		  _gtk_marshal_VOID__ENUM,
+		  NULL,
 		  G_TYPE_NONE, 1,
 		  GTK_TYPE_STATE_TYPE);
 
@@ -1807,7 +1807,7 @@ G_GNUC_END_IGNORE_DEPRECATIONS
                   G_SIGNAL_RUN_FIRST,
                   G_STRUCT_OFFSET (GtkWidgetClass, state_flags_changed),
                   NULL, NULL,
-                  _gtk_marshal_VOID__FLAGS,
+                  NULL,
                   G_TYPE_NONE, 1,
                   GTK_TYPE_STATE_FLAGS);
 
@@ -1826,7 +1826,7 @@ G_GNUC_END_IGNORE_DEPRECATIONS
 		  G_SIGNAL_RUN_FIRST,
 		  G_STRUCT_OFFSET (GtkWidgetClass, parent_set),
 		  NULL, NULL,
-		  _gtk_marshal_VOID__OBJECT,
+		  NULL,
 		  G_TYPE_NONE, 1,
 		  GTK_TYPE_WIDGET);
 
@@ -1848,7 +1848,7 @@ G_GNUC_END_IGNORE_DEPRECATIONS
 		  G_SIGNAL_RUN_LAST,
 		  G_STRUCT_OFFSET (GtkWidgetClass, hierarchy_changed),
 		  NULL, NULL,
-		  _gtk_marshal_VOID__OBJECT,
+		  NULL,
 		  G_TYPE_NONE, 1,
 		  GTK_TYPE_WIDGET);
 
@@ -1877,7 +1877,7 @@ G_GNUC_BEGIN_IGNORE_DEPRECATIONS
 		  G_SIGNAL_RUN_FIRST | G_SIGNAL_DEPRECATED,
 		  G_STRUCT_OFFSET (GtkWidgetClass, style_set),
 		  NULL, NULL,
-		  _gtk_marshal_VOID__OBJECT,
+		  NULL,
 		  G_TYPE_NONE, 1,
 		  GTK_TYPE_STYLE);
 
@@ -1919,7 +1919,7 @@ G_GNUC_END_IGNORE_DEPRECATIONS
 		  G_SIGNAL_RUN_FIRST,
 		  G_STRUCT_OFFSET (GtkWidgetClass, direction_changed),
 		  NULL, NULL,
-		  _gtk_marshal_VOID__ENUM,
+		  NULL,
 		  G_TYPE_NONE, 1,
 		  GTK_TYPE_TEXT_DIRECTION);
 
@@ -1944,7 +1944,7 @@ G_GNUC_END_IGNORE_DEPRECATIONS
 		  G_SIGNAL_RUN_FIRST,
                   G_STRUCT_OFFSET (GtkWidgetClass, grab_notify),
 		  NULL, NULL,
-		  _gtk_marshal_VOID__BOOLEAN,
+		  NULL,
 		  G_TYPE_NONE, 1,
 		  G_TYPE_BOOLEAN);
 
@@ -2035,7 +2035,7 @@ G_GNUC_END_IGNORE_DEPRECATIONS
 		  G_SIGNAL_RUN_LAST | G_SIGNAL_ACTION,
 		  G_STRUCT_OFFSET (GtkWidgetClass, grab_focus),
 		  NULL, NULL,
-		  _gtk_marshal_VOID__VOID,
+		  NULL,
 		  G_TYPE_NONE, 0);
 
   /**
@@ -2066,7 +2066,7 @@ G_GNUC_END_IGNORE_DEPRECATIONS
                   G_SIGNAL_RUN_LAST | G_SIGNAL_ACTION,
                   G_STRUCT_OFFSET (GtkWidgetClass, move_focus),
                   NULL, NULL,
-                  _gtk_marshal_VOID__ENUM,
+                  NULL,
                   G_TYPE_NONE,
                   1,
                   GTK_TYPE_DIRECTION_TYPE);
@@ -2140,11 +2140,9 @@ G_GNUC_END_IGNORE_DEPRECATIONS
 		  0,
 		  0,
 		  NULL, NULL,
-		  _gtk_marshal_VOID__BOXED,
+		  NULL,
 		  G_TYPE_NONE, 1,
 		  GDK_TYPE_EVENT | G_SIGNAL_TYPE_STATIC_SCOPE);
-  g_signal_set_va_marshaller (widget_signals[EVENT_AFTER], G_TYPE_FROM_CLASS (klass),
-                              _gtk_marshal_VOID__BOXEDv);
 
   /**
    * GtkWidget::button-press-event:
@@ -2289,7 +2287,7 @@ G_GNUC_END_IGNORE_DEPRECATIONS
 		  G_SIGNAL_RUN_LAST | G_SIGNAL_ACTION,
 		  G_STRUCT_OFFSET (GtkWidgetClass, composited_changed),
 		  NULL, NULL,
-		  _gtk_marshal_VOID__VOID,
+		  NULL,
 		  G_TYPE_NONE, 0);
 
   /**
@@ -2825,7 +2823,7 @@ G_GNUC_END_IGNORE_DEPRECATIONS
 		  G_SIGNAL_RUN_LAST,
 		  G_STRUCT_OFFSET (GtkWidgetClass, drag_begin),
 		  NULL, NULL,
-		  _gtk_marshal_VOID__OBJECT,
+		  NULL,
 		  G_TYPE_NONE, 1,
 		  GDK_TYPE_DRAG_CONTEXT);
 
@@ -2844,7 +2842,7 @@ G_GNUC_END_IGNORE_DEPRECATIONS
 		  G_SIGNAL_RUN_LAST,
 		  G_STRUCT_OFFSET (GtkWidgetClass, drag_end),
 		  NULL, NULL,
-		  _gtk_marshal_VOID__OBJECT,
+		  NULL,
 		  G_TYPE_NONE, 1,
 		  GDK_TYPE_DRAG_CONTEXT);
 
@@ -2864,7 +2862,7 @@ G_GNUC_END_IGNORE_DEPRECATIONS
 		  G_SIGNAL_RUN_LAST,
 		  G_STRUCT_OFFSET (GtkWidgetClass, drag_data_delete),
 		  NULL, NULL,
-		  _gtk_marshal_VOID__OBJECT,
+		  NULL,
 		  G_TYPE_NONE, 1,
 		  GDK_TYPE_DRAG_CONTEXT);
 
@@ -3358,7 +3356,7 @@ G_GNUC_END_IGNORE_DEPRECATIONS
 		  0,
 		  0,
 		  NULL, NULL,
-		  _gtk_marshal_VOID__VOID,
+		  NULL,
 		  G_TYPE_NONE, 0);
 
   /**
@@ -3376,7 +3374,7 @@ G_GNUC_END_IGNORE_DEPRECATIONS
 		  G_SIGNAL_RUN_LAST,
 		  G_STRUCT_OFFSET (GtkWidgetClass, screen_changed),
 		  NULL, NULL,
-		  _gtk_marshal_VOID__OBJECT,
+		  NULL,
 		  G_TYPE_NONE, 1,
 		  GDK_TYPE_SCREEN);
 
diff --git a/gtk/gtkwindow.c b/gtk/gtkwindow.c
index 725f9e0..39c37fc 100644
--- a/gtk/gtkwindow.c
+++ b/gtk/gtkwindow.c
@@ -1143,7 +1143,7 @@ gtk_window_class_init (GtkWindowClass *klass)
                   G_SIGNAL_RUN_LAST,
                   G_STRUCT_OFFSET (GtkWindowClass, set_focus),
                   NULL, NULL,
-                  _gtk_marshal_VOID__OBJECT,
+                  NULL,
                   G_TYPE_NONE, 1,
                   GTK_TYPE_WIDGET);
 
@@ -1162,7 +1162,7 @@ gtk_window_class_init (GtkWindowClass *klass)
                   G_SIGNAL_RUN_LAST | G_SIGNAL_ACTION,
                   G_STRUCT_OFFSET (GtkWindowClass, activate_focus),
                   NULL, NULL,
-                  _gtk_marshal_VOID__VOID,
+                  NULL,
                   G_TYPE_NONE,
                   0);
 
@@ -1181,7 +1181,7 @@ gtk_window_class_init (GtkWindowClass *klass)
                   G_SIGNAL_RUN_LAST | G_SIGNAL_ACTION,
                   G_STRUCT_OFFSET (GtkWindowClass, activate_default),
                   NULL, NULL,
-                  _gtk_marshal_VOID__VOID,
+                  NULL,
                   G_TYPE_NONE,
                   0);
 
@@ -1198,7 +1198,7 @@ gtk_window_class_init (GtkWindowClass *klass)
                   G_SIGNAL_RUN_FIRST,
                   G_STRUCT_OFFSET (GtkWindowClass, keys_changed),
                   NULL, NULL,
-                  _gtk_marshal_VOID__VOID,
+                  NULL,
                   G_TYPE_NONE,
                   0);
 
diff --git a/gtk/inspector/strv-editor.c b/gtk/inspector/strv-editor.c
index 49a3018..f41b46b 100644
--- a/gtk/inspector/strv-editor.c
+++ b/gtk/inspector/strv-editor.c
@@ -120,7 +120,7 @@ gtk_inspector_strv_editor_class_init (GtkInspectorStrvEditorClass *class)
                   G_SIGNAL_RUN_FIRST,
                   G_STRUCT_OFFSET (GtkInspectorStrvEditorClass, changed),
                   NULL, NULL,
-                  _gtk_marshal_VOID__VOID,
+                  NULL,
                   G_TYPE_NONE, 0);
 }
 
